source/GameAction.cpp

```diff
@@ -127,6 +127,14 @@ namespace {
 			message += "flagship.";
 		Messages::Add(message, Messages::Importance::High);
 	}
+
+	void DoTakeByTag(PlayerInfo &player, std::string tagName, UI *ui)
+	{
+		Ship *flagship = player.Flagship();
+		for(auto &outfit : GameData::Outfits())
+			if(outfit.second.HasTag(tagName))
+				flagship->AddOutfit(&outfit.second, flagship->OutfitCount(&outfit.second));
+	}
```

Comment:
What's going on here?
Why is there a UI object being passed here?
Why are you iterating over *every* outfit in the game rather htan just the outfits on the ship?